### PR TITLE
Fix inconsistent error behavior

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- restore `serializeError` 
- wrap `provider.request` to catch error and serialize to the same error format
- nit: fix error type for invalid arg check

https://github.com/coinbase/coinbase-wallet-sdk/blob/965119c7a66bb5b8f94bfc47cb6813f260e40dd8/packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts#L471

### _How did you test your changes?_

existing tests

### CODE REVIEW

File: packages/wallet-sdk/src/core/error/errors.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { errorValues, standardErrorCodes } from './constants';
+import { standardErrorCodes } from './constants';
+import { getMessageFromCode } from './utils';
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/error/errors.ts:2]: https://github.cbhq.net/wallet/sdk-internal/pull/109

File: packages/wallet-sdk/src/core/error/utils.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+function isErrorWithCode(error: unknown): error is ErrorWithCode {
+  return (
+    typeof error === 'object' &&
+    error !== null &&
+    (typeof (error as ErrorWithCode).code === 'number' ||
+      typeof (error as ErrorWithCode).errorCode === 'number')
+  );
+}
+
+export function serialize(
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/error/utils.ts:78]: https://github.cbhq.net/wallet/sdk-internal/pull/109

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 3]
```diff
@@ -52,11 +53,15 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
   }
 
   public async request<T>(args: RequestArguments): Promise<T> {
-    const invalidArgsError = checkErrorForInvalidRequestArgs(args);
-    if (invalidArgsError) throw invalidArgsError;
-    // unrecognized methods are treated as fetch requests
-    const category = determineMethodCategory(args.method) ?? 'fetch';
-    return this.handlers[category](args) as T;
+    try {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:56]: https://github.com/coinbase/coinbase-wallet-sdk/pull/1285/files?diff=split&w=1

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 4]
```diff
@@ -52,11 +53,15 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
...
+      const invalidArgsError = checkErrorForInvalidRequestArgs(args);
+      if (invalidArgsError) throw invalidArgsError;
+      // unrecognized methods are treated as fetch requests
+      const category = determineMethodCategory(args.method) ?? 'fetch';
+      return this.handlers[category](args) as T;
+    } catch (error) {
+      return Promise.reject(serializeError(error));
+    }
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:64]: https://github.com/coinbase/coinbase-wallet-sdk/blob/965119c7a66bb5b8f94bfc47cb6813f260e40dd8/packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts#L477

File: packages/wallet-sdk/src/core/error/errors.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
-import { errorValues, standardErrorCodes } from './constants';
+import { standardErrorCodes } from './constants';
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/error/errors.ts:1]: reverting https://github.cbhq.net/wallet/sdk-internal/pull/109

**Review by eric-ships — APPROVED** [review]
Looks good, appreciate the quality

**Comment by cb-heimdall** [comment]
Review Error for eric-ships @ 2024-05-09 23:33:30 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by eric-ships — APPROVED** [review]

**Pull request merged**